File: algorithm-design/leetcode/lc029_divide_two_integers/LC029DivideTwoIntegers.Tests.cs
using System;
using Xunit;

namespace AlgorithmDesign.leetcode.lc029_divide_two_integers
{
    public class SolutionTests
    {
        [Fact]
        public void test_Minus1010369383_Minus2147483648()
        {
            Assert.Equal(-1010369383 / -2147483648, new Solution().Divide(-1010369383, -2147483648));
        }

        [Fact]
        public void test_Minus2147483648_Minus1()
        {
            Assert.Equal(Int32.MaxValue, new Solution().Divide(-2147483648, -1));
        }

        [Fact]
        public void test_divisor_0()
        {
            Assert.Equal(Int32.MaxValue, new Solution().Divide(1, 0));
        }

        [Fact]
        public void test_dividend_0()
        {
            Assert.Equal(0, new Solution().Divide(0, 1));
        }
    }
}

File: algorithms/code/leetcode/lc029_divide_two_integers/LC029DivideTwoIntegersTests.java
package leetcode.lc029_divide_two_integers;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public final class LC029DivideTwoIntegersTests {
    @Test
    public void testMinus1010369383Minus2147483648() throws Exception {
        LC029DivideTwoIntegers solution;
        solution = new LC029DivideTwoIntegers();
        assertEquals(-1010369383 / -2147483648, solution.divide(-1010369383, -2147483648));
    }

    @Test
    public void testMinus2147483648Minus1() throws Exception {
        LC029DivideTwoIntegers solution;
        solution = new LC029DivideTwoIntegers();
        assertEquals(Integer.MAX_VALUE, solution.divide(-2147483648, -1));
    }

    @Test
    public void testDivisor0() throws Exception {
        LC029DivideTwoIntegers solution;
        solution = new LC029DivideTwoIntegers();
        assertEquals(Integer.MAX_VALUE, solution.divide(1, 0));
    }

    @Test
    public void testDividend0() throws Exception {
        LC029DivideTwoIntegers solution;
        solution = new LC029DivideTwoIntegers();
        assertEquals(0, solution.divide(0, 1));
    }
}
